File: lib/Models/project.dart
class Project {
  //step1 ----> here I will Create the class's variables :
  final int id;
  final String name;
  final String description;
  final String techStack;
  final String githubLink;

  //step2 ----> here I will Create The constructor :
  Project({
    required this.id,
    required this.name,
    required this.description,
    required this.techStack,
    required this.githubLink,
  });

  //step3 ----> here I will Create Methode to transfer from Json :
  factory Project.fromJson(Map<String, dynamic> json) {
    return Project(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      techStack: json['techStack'],
      githubLink: json['githubLink'],
    );
  }
}

File: Backend/src/main/java/com/apps/PortfolioApp/Model/Entity/Project.java
package com.apps.PortfolioApp.Model.Entity;

import jakarta.persistence.*;

@Table(name = "projects")
@Entity
public class Project {
    //  here I will Create Class variables :
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id ;
    private String name;
    @Column(length = 500)
    private String description;
    @Column(name = "tech_stack")
    private String techStack;
    @Column(name = "github_link")
    private String githubLink;

    // here I will Create a constructor :
    public Project() {}

    public Project(Integer id, String name, String description, String techStack, String githubLink) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.techStack = techStack;
        this.githubLink = githubLink;
    }

    // Getter && Setter
    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getTechStack() {
        return techStack;
    }

    public void setTechStack(String techStack) {
        this.techStack = techStack;
    }

    public String getGithubLink() {
        return githubLink;
    }

    public void setGithubLink(String githubLink) {
        this.githubLink = githubLink;
    }
}